src/types/Profiles.cpp

```diff
@@ -1,38 +1,46 @@
 #include "../../include/types/Profiles.h"
+#include "../../include/types/Perf.h"
+#include "../../include/types/Performance.h"
 #include "../../include/types/PerformanceTable.h"
 #include "../../include/utils.h"
 #include <algorithm>
 #include <iostream>
+#include <random>
+#include <sstream>
 #include <string>
 #include <vector>
 
 Profiles::Profiles(std::vector<Performance> &perf_vect)
     : PerformanceTable(perf_vect) {
-  if (!this->isProfileOrdered()) {
-    throw std::invalid_argument(
-        "The given performance vector cannot be used as a profiles performance "
-        "table. Each row must be ranked such that for each row i we have on "
-        "each criterion j : v_i - 1_j > v_i_j > v_i + 1_j");
-  }
+  mode_ = "crit";
```

Comment:
It could be a great time to start using logs here: let's log a warning if the profile is not in an ordered state, so we can keep a trace of that later !

Reply:
lets do logging after :)